open_samus_returns_rando/pickup.py

```diff
@@ -1,53 +1,37 @@
 import copy
 import logging
-from pathlib import Path
+from enum import Enum
 
 from mercury_engine_data_structures.formats import Bmsad
 
+from open_samus_returns_rando.lua_editor import LuaEditor
 from open_samus_returns_rando.model_data import get_data
 from open_samus_returns_rando.patcher_editor import PatcherEditor, path_for_level
 
 LOG = logging.getLogger("pickup")
 
+class PickupType(Enum):
+    ACTOR = "actor"
 
-def _read_powerup_lua() -> bytes:
-    return Path(__file__).parent.joinpath("files", "randomizer_powerup.lua").read_bytes()
+class BasePickup:
+    def __init__(self, lua_editor: LuaEditor, pickup: dict, pickup_id: int, configuration: dict):
+        self.lua_editor = lua_editor
+        self.pickup = pickup
+        self.pickup_id = pickup_id
+        self.configuration = configuration
 
+    def patch(self, editor: PatcherEditor):
+        raise NotImplementedError
 
-def patch_pickups(editor: PatcherEditor, pickups_config: list[dict]):
-    template_bmsad = editor.get_parsed_asset("actors/items/powerup_chargebeam/charclasses/powerup_chargebeam.bmsad").raw
+# Note: MSR has a lot of customised Lua for Pickups. We maybe need to add them / take the logic into account?
```

Comment:
If this is talking about the pickups folder, i think we would just remove those and it would be included in `randomizer_powerup.lua`. I only had those files to temporarily test and modify certain pickups.